runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -473,18 +477,23 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
          }
 
       clientSession->getSequencingMonitor()->enter();
-      // Update the expecting sequence number. This will allow subsequent
-      // threads to pass through once we've released the sequencing monitor.
-      TR_ASSERT(seqNo == clientSession->getExpectedSeqNo(), "Unexpected seqNo=%u expected=%u\n", seqNo, clientSession->getExpectedSeqNo());
 
-      //TODO: Might need to set the expectedSeqNo as (seqNo+1) instead of (expectedSeqNo+1)
       if (seqNo != clientSession->getExpectedSeqNo())
+         {
          Trc_JITServerUnexpectedSeqNo(compThread, getCompThreadId(), clientSession,
                (unsigned long long)clientId, seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
 
+         TR_ASSERT_FATAL(false, "compThreadID=%d clientSessionData=%p clientUID=%llu (seqNo=%u, expectedSeqNo=%u, numActiveThreads=%d) unexpected seqNo",
+               getCompThreadId(), clientSession, (unsigned long long)clientId,
+               seqNo, clientSession->getExpectedSeqNo(), clientSession->getNumActiveThreads());
+
+         clientSession->setExpectedSeqNo(seqNo);
```

Comment:
This acts like a correction, but it's never executed because of the fatal assert above. Is the intent to transform the fatal assert into an simple assert?

Reply:
I'm not sure if we should have a FATAL assert here to catch the issue in a release build, or make it a simple assert which means if the sequence of messages are not handled properly we might not catch it right away. But a simple assert is also less intrusive.

Reply:
If we can safely recover from a bug I prefer to have the soft assert.